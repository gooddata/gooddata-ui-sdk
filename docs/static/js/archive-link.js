document.addEventListener('DOMContentLoaded', function () {
    function createArchiveLink() {
        const link = document.createElement('a');
        link.className = 'dropdown-item';
        
        const currentHostname = window.location.hostname;
        if (currentHostname.includes('netlify.app')) {
            // For Netlify previews, link to the versions page within the current preview site
            link.href = window.location.origin + '/docs/gooddata-ui/versions/';
        } else if (currentHostname === 'localhost' || currentHostname === '127.0.0.1') {
            // For local testing, link to the versions page
            link.href = '/versions/';
        } else {
            // For production - link to the unversioned archive page
            link.href = 'https://www.gooddata.com/docs/gooddata-ui/versions/';
        }
        
        // Safe DOM manipulation - XSS protected
        const icon = document.createElement('i');
        icon.className = 'fas fa-archive me-2';
        icon.setAttribute('aria-hidden', 'true');
        
        link.appendChild(icon);
        link.appendChild(document.createTextNode('Archive'));
        
        // Add descriptive text for screen readers
        link.setAttribute('aria-label', 'View all versions in archive');
        
        return link;
    }
    
    // Select the dropdown menu and add the archive link
    const dropdownMenus = document.getElementsByClassName('dropdown-menu');
    for (let i = 0; i < dropdownMenus.length; i++) {
        const archiveLink = createArchiveLink();
        dropdownMenus[i].appendChild(archiveLink);
    }
});