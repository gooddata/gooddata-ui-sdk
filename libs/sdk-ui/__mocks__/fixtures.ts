// (C) 2007-2019 GoodData Corporation
import { LegacyExecutionRecording, legacyRecordedDataView } from "@gooddata/sdk-backend-mockingbird";
import { IInsight } from "@gooddata/sdk-model";
import { DataViewFacade } from "../src/base/results/facade";

export const testWorkspace = "testWorkspace";

function legacyRecordedDataFacade(rec: LegacyExecutionRecording): DataViewFacade {
    return DataViewFacade.for(legacyRecordedDataView(rec));
}

const BarChartForDrillTests = {
    definition: require("./recordings/bar_chart_for_drill_tests/definition.json"),
    response: require("./recordings/bar_chart_for_drill_tests/response.json"),
    result: require("./recordings/bar_chart_for_drill_tests/result.json"),
};

export const barChartForDrillTests = legacyRecordedDataFacade(BarChartForDrillTests);

export const insightWithPoP: IInsight = {
    insight: {
        identifier: "popMeasureInsight",
        uri: "/some/uri/popMeasureInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier: "m1",
                            title: "# Accounts with AD Query",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_pop",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "m1",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_previous_period",
                            definition: {
                                previousPeriodMeasure: {
                                    measureIdentifier: "m1",
                                    dateDataSets: [
                                        {
                                            dataSet: {
                                                uri: "/gdc/md/myproject/obj/921",
                                            },
                                            periodsAgo: 1,
                                        },
                                    ],
                                },
                            },
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "PoP measure",
        sorts: [],
        properties: {},
    },
};

export const insightWithPoPAndAlias: IInsight = {
    insight: {
        identifier: "popMeasureWithAliasInsight",
        uri: "/some/uri/popMeasureInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier: "m1",
                            title: "# Accounts with AD Query",
                            alias: "AD Queries",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_pop",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "m1",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_previous_period",
                            definition: {
                                previousPeriodMeasure: {
                                    measureIdentifier: "m1",
                                    dateDataSets: [
                                        {
                                            dataSet: {
                                                uri: "/gdc/md/myproject/obj/921",
                                            },
                                            periodsAgo: 1,
                                        },
                                    ],
                                },
                            },
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "PoP measure with alias",
        sorts: [],
        properties: {},
    },
};

export const insightWithArithmeticAndDerivedMeasures: IInsight = {
    insight: {
        identifier: "arithmeticAndDerivedMeasureInsight",
        uri: "/some/uri/arithmeticAndDerivedMeasureInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier: "am1",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "m1_pop"],
                                },
                            },
                            title: "ignored title",
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1",
                            title: "AD Accounts",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_pop",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "m1",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                            title: "ignored title",
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "Arithmetic and derived measures",
        sorts: [],
        properties: {},
    },
};

export const insightWithArithmeticMeasureTree: IInsight = {
    insight: {
        identifier: "arithmeticMeasureTreeInsight",
        uri: "/some/uri/arithmeticMeasureTreeInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier:
                                "arithmetic_measure_created_from_complicated_arithmetic_measures",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: [
                                        "arithmetic_measure_created_from_arithmetic_measures",
                                        "arithmetic_measure_created_from_simple_measures",
                                    ],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1",
                            title: "AD Accounts",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m2",
                            title: "KD Accounts",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/1245",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "tree_level_2",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "tree_level_1"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "tree_level_1",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "tree_level_0"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "tree_level_0",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "m2"],
                                },
                            },
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "Arithmetic measure tree",
        sorts: [],
        properties: {},
    },
};

export const insightWithComplexArithmeticMeasureTree: IInsight = {
    insight: {
        identifier: "complexArithmeticMeasureTreeInsight",
        uri: "/some/uri/complexArithmeticMeasureTreeInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier:
                                "arithmetic_measure_created_from_complicated_arithmetic_measures",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: [
                                        "arithmetic_measure_created_from_arithmetic_measures",
                                        "arithmetic_measure_created_from_simple_measures",
                                    ],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1",
                            title: "# Accounts with AD Query",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m2",
                            title: "# Accounts with KD Query",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/1245",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m3",
                            title: "# Accounts with AD Query",
                            alias: "AD Queries",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/8172",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m4",
                            title: "# Accounts with KD Query",
                            alias: "KD Queries",
                            definition: {
                                measureDefinition: {
                                    item: {
                                        uri: "/gdc/md/myproject/obj/1245",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_pop",
                            title: "M1 Pop Measure Title",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "m1",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_previous_period",
                            title: "M1 Previous Measure Title",
                            definition: {
                                previousPeriodMeasure: {
                                    measureIdentifier: "m1",
                                    dateDataSets: [
                                        {
                                            dataSet: {
                                                uri: "/gdc/md/myproject/obj/921",
                                            },
                                            periodsAgo: 1,
                                        },
                                    ],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_pop_renamed",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "m1",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                            alias: "Renamed SP last year M1",
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "m1_previous_period_renamed",
                            definition: {
                                previousPeriodMeasure: {
                                    measureIdentifier: "m1",
                                    dateDataSets: [
                                        {
                                            dataSet: {
                                                uri: "/gdc/md/myproject/obj/921",
                                            },
                                            periodsAgo: 1,
                                        },
                                    ],
                                },
                            },
                            alias: "Renamed previous period M1",
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "derived_measure_from_arithmetic_measure",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "arithmetic_measure_created_from_simple_measures",
                                    popAttribute: {
                                        uri: "/gdc/md/myproject/obj/1514",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "arithmetic_measure_created_from_simple_measures",
                            title: "Arithmetic sum(m1, m2) Measure Title",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "m2"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "arithmetic_measure_created_from_renamed_simple_measures",
                            title: "Arithmetic sum(m3, m4) Measure Title",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m3", "m4"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "arithmetic_measure_created_from_derived_measures",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1_pop", "m1_previous_period"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "arithmetic_measure_created_from_arithmetic_measures",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: [
                                        "arithmetic_measure_created_from_renamed_simple_measures",
                                        "arithmetic_measure_created_from_renamed_derived_measures",
                                    ],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "arithmetic_measure_created_from_renamed_derived_measures",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1_pop_renamed", "m1_previous_period_renamed"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "invalid_arithmetic_measure_with_missing_dependency",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: ["m1", "m666"],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "invalid_arithmetic_measure_with_cyclic_dependency_1",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: [
                                        "m1",
                                        "invalid_arithmetic_measure_with_cyclic_dependency_2",
                                    ],
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "invalid_arithmetic_measure_with_cyclic_dependency_2",
                            definition: {
                                arithmeticMeasure: {
                                    operator: "sum",
                                    measureIdentifiers: [
                                        "m2",
                                        "invalid_arithmetic_measure_with_cyclic_dependency_1",
                                    ],
                                },
                            },
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "Complex arithmetic measure tree",
        sorts: [],
        properties: {},
    },
};

export const insightWithMultipleMeasureBuckets: IInsight = {
    insight: {
        identifier: "multipleMeasureBucketsInsight",
        uri: "/some/uri/multipleMeasureBucketsInsight",
        visualizationUrl: "local:test",
        buckets: [
            {
                localIdentifier: "measures",
                items: [
                    {
                        measure: {
                            localIdentifier: "fdd41e4ca6224cd2b5ecce15fdabf062",
                            format: "#,##0.00",
                            title: "Sum of Email Clicks",
                            definition: {
                                measureDefinition: {
                                    aggregation: "sum",
                                    item: {
                                        uri: "/gdc/md/yrungi0zwpoud7h1kmh6ldhp0vgkpi41/obj/15428",
                                    },
                                },
                            },
                        },
                    },
                ],
            },
            {
                localIdentifier: "secondary_measures",
                items: [
                    {
                        measure: {
                            localIdentifier: "fdd41e4ca6224cd2b5ecce15fdabf062_pop",
                            definition: {
                                popMeasureDefinition: {
                                    measureIdentifier: "fdd41e4ca6224cd2b5ecce15fdabf062",
                                    popAttribute: {
                                        uri: "/gdc/md/yrungi0zwpoud7h1kmh6ldhp0vgkpi41/obj/15330",
                                    },
                                },
                            },
                        },
                    },
                    {
                        measure: {
                            localIdentifier: "fdd41e4ca6224cd2b5ecce15fdabf062_previous_period",
                            definition: {
                                previousPeriodMeasure: {
                                    measureIdentifier: "fdd41e4ca6224cd2b5ecce15fdabf062",
                                    dateDataSets: [
                                        {
                                            dataSet: {
                                                uri: "/gdc/md/yrungi0zwpoud7h1kmh6ldhp0vgkpi41/obj/921",
                                            },
                                            periodsAgo: 1,
                                        },
                                    ],
                                },
                            },
                        },
                    },
                ],
            },
        ],
        filters: [],
        title: "Multiple measure buckets",
        sorts: [],
        properties: {},
    },
};
