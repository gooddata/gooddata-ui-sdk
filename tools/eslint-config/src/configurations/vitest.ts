// (C) 2025 GoodData Corporation

import type { IConfiguration } from "../types.js";

const configuration: IConfiguration<"@vitest"> = {
    packages: [
        {
            name: "@vitest/eslint-plugin",
            version: "1.3.5",
        },
    ],
    plugin: "@vitest",
    extends: ["plugin:@vitest/legacy-recommended"],
    rules: {
        "@vitest/expect-expect": "off",
        "@vitest/no-commented-out-tests": "warn",
        "@vitest/valid-title": "warn",
        "@vitest/no-disabled-tests": "warn",
        "@vitest/no-focused-tests": "warn",
        "@vitest/no-identical-title": "warn",
        "@vitest/valid-expect": "warn",
    },
};

export default configuration;
