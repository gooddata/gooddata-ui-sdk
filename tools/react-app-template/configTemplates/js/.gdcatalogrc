{
  "hostname": "https://public-examples.gooddata.com",
  "workspaceId": "demo",
  "output": "src/catalog.ts",
  "backend": "tiger",
  "demo": true
}