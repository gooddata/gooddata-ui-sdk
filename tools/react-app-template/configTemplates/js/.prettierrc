{
    "printWidth": 110,
    "tabWidth": 4,
    "trailingComma": "all"
}
