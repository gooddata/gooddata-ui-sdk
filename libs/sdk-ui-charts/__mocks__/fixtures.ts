// (C) 2007-2019 GoodData Corporation
import { LegacyExecutionRecording, legacyRecordedDataView } from "@gooddata/sdk-backend-mockingbird";

import * as rec from "./recordings/playlist.js";
import { DataViewFacade } from "@gooddata/sdk-ui";

function legacyRecordedDataFacade(rec: LegacyExecutionRecording): DataViewFacade {
    return DataViewFacade.for(legacyRecordedDataView(rec));
}

//
// new fixtures
//

export const testWorkspace = "testWorkspace";

//
// Area chart fixtures
//

export const areaChartWith3MetricsAndViewByAttribute = legacyRecordedDataFacade(
    rec.AreaChartWith3MetricsAndViewByAttribute,
);

export const areaChartWithMeasureViewByAndStackBy = legacyRecordedDataFacade(
    rec.AreaChartWithMeasureViewByAndStackBy,
);

//
// Bar chart fixtures
//

export const barChartWith4MetricsAndViewByTwoAttributes = legacyRecordedDataFacade(
    rec.BarChartWith4MetricsAndViewByTwoAttributes,
);
export const barChartWithStackByAndViewByAttributes = legacyRecordedDataFacade(
    rec.BarChartWithStackByAndViewByAttributes,
);
export const barChartWithViewByAttribute = legacyRecordedDataFacade(rec.BarChartWithViewByAttribute);

export const barChartWith3MetricsAndViewByAttribute = legacyRecordedDataFacade(
    rec.BarChartWith3MetricsAndViewByAttribute,
);
export const barChartWith3MetricsAndViewByAttributeFunformat = legacyRecordedDataFacade(
    rec.BarChartWith3MetricsAndViewByAttributeFunformat,
);

export const barChartWith3MetricsAndViewByAttributePercInFormat = legacyRecordedDataFacade(
    rec.BarChartWith3MetricsAndViewByAttributePercInFormat,
);

export const barChartWithPopMeasureAndViewByAttribute = legacyRecordedDataFacade(
    rec.BarChartWithPopMeasureAndViewByAttribute,
);

export const barChartWithPopMeasureAndViewByAttributeX6 = legacyRecordedDataFacade(
    rec.BarChartWithPopMeasureAndViewByAttributeX6,
);

export const barChartWithPreviousPeriodMeasure = legacyRecordedDataFacade(
    rec.BarChartWithPreviousPeriodMeasure,
);
export const barChartWithSingleMeasureAndNoAttributes = legacyRecordedDataFacade(
    rec.BarChartWithSingleMeasureAndNoAttributes,
);

export const barChartWithPreviousPeriodMeasureX6 = legacyRecordedDataFacade(
    rec.BarChartWithPreviousPeriodMeasureX6,
);

//
// Bubble chart
//
export const bubbleChartWith2MetricsAndAttributeNoPrimaries = legacyRecordedDataFacade(
    rec.BubbleChartWith2MetricsAndAttributeNoPrimaries,
);

export const bubbleChartWith1Metric = legacyRecordedDataFacade(rec.BubbleChartWith1Metric);

export const bubbleChartWith3MetricsAndAttribute = legacyRecordedDataFacade(
    rec.BubbleChartWith3MetricsAndAttribute,
);
export const bubbleChartWith3MetricsAndAttributeNullsInData = legacyRecordedDataFacade(
    rec.BubbleChartWith3MetricsAndAttributeNullsInData,
);

//
// Combo chart
//
export const comboWithTwoMeasuresAndViewByAttribute = legacyRecordedDataFacade(
    rec.ComboChartWithTwoMeasuresViewByAttribute,
);

export const comboChartWithTwoMeasuresViewByAttributeNoBuckets = legacyRecordedDataFacade(
    rec.ComboChartWithTwoMeasuresViewByAttributeNoBuckets,
);

export const comboChartWithTwoMeasuresViewByAttributePercformat = legacyRecordedDataFacade(
    rec.ComboChartWithTwoMeasuresViewByAttributePercformat,
);

//
//
//
export const headlineWithOneMeasure = legacyRecordedDataFacade(rec.HeadlineWithOneMeasure);
export const headlineWithOneMeasureWithIdentifier = legacyRecordedDataFacade(
    rec.HeadlineWithOneMeasureWithIdentifier,
);
export const headlineWithTwoMeasures = legacyRecordedDataFacade(rec.HeadlineWithTwoMeasures);
export const headlineWithTwoMeasuresWithIdentifier = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresWithIdentifier,
);
export const headlineWithTwoMeasuresFirstEmpty = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresFirstEmpty,
);
export const headlineWithTwoMeasuresSecondEmpty = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresSecondEmpty,
);
export const headlineWithTwoMeasuresBothEmpty = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresBothEmpty,
);
export const headlineWithTwoMeasuresBothZero = legacyRecordedDataFacade(rec.HeadlineWithTwoMeasuresBothZero);
export const headlineWithTwoMeasuresFirstZero = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresFirstZero,
);
export const headlineWithTwoMeasuresSecondZero = legacyRecordedDataFacade(
    rec.HeadlineWithTwoMeasuresSecondZero,
);
export const headlineWithTwoMeasuresBothSame = legacyRecordedDataFacade(rec.HeadlineWithTwoMeasuresBothSame);

//
// Heatmap recordings
//

export const heatMapWithEmptyCells = legacyRecordedDataFacade(rec.HeatMapWithEmptyCells);
export const heatMapWithMetricRowColumn = legacyRecordedDataFacade(rec.HeatMapWithMetricRowColumn);
//
// Pie chart recordings
//

export const pieChartWithMetricsOnly = legacyRecordedDataFacade(rec.PieChartWithMetricsOnly);
export const pieChartWithMetricsOnlyFundata = legacyRecordedDataFacade(rec.PieChartWithMetricsOnlyFundata);

//
// Scatter plot recording
//

export const scatterPlotWith2MetricsAndAttributeNullsInData = legacyRecordedDataFacade(
    rec.ScatterPlotWith2MetricsAndAttributeNullsInData,
);
export const scatterPlotWith2MetricsAndAttributeWithPrimary = legacyRecordedDataFacade(
    rec.ScatterPlotWith2MetricsAndAttributeWithPrimary,
);

//
// Treemap recordings
//

export const treemapWithMetricAndStackByAttribute = legacyRecordedDataFacade(
    rec.TreemapWithMetricAndStackByAttribute,
);
export const treemapWithMetricAndViewByAttribute = legacyRecordedDataFacade(
    rec.TreemapWithMetricAndViewByAttribute,
);
export const treemapWithTwoMetricsAndStackByAttribute = legacyRecordedDataFacade(
    rec.TreemapWithTwoMetricsAndStackByAttribute,
);
export const treemapWithMetricViewByAndStackByAttribute = legacyRecordedDataFacade(
    rec.TreemapWithMetricViewByAndStackByAttribute,
);
