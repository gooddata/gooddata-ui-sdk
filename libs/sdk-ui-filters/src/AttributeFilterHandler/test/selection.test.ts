// (C) 2019-2022 GoodData Corporation
import { newTestAttributeFilterHandler } from "./fixtures.js";
import { describe, it, expect, vi } from "vitest";

const changedSelection = {
    isInverted: true,
    keys: ["/selectedElement"],
    irrelevantKeys: [] as string[],
};

const changedSelectionWithIrrelevantElements = {
    isInverted: true,
    keys: ["/selectedElement", "/irrelevantElement"],
    irrelevantKeys: ["/irrelevantElement"],
};

describe("MultiSelectAttributeFilterHandler", () => {
    it("should initialize selection for positive attribute filter", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        expect(attributeFilterHandler.getCommittedSelection()).toMatchSnapshot("committed selection");
        expect(attributeFilterHandler.getWorkingSelection()).toMatchSnapshot("working selection");
    });

    it("should initialize selection for negative attribute filter", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("negative");

        expect(attributeFilterHandler.getCommittedSelection()).toMatchSnapshot("committed selection");
        expect(attributeFilterHandler.getWorkingSelection()).toMatchSnapshot("working selection");
    });

    it("changeSelection() should trigger onSelectionChanged() callback", async () => {
        const onSelectionChanged = vi.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onSelectionChanged(onSelectionChanged);
        attributeFilterHandler.changeSelection(changedSelection);

        expect(onSelectionChanged).toHaveBeenCalledTimes(1);
        expect(onSelectionChanged.mock.calls[0]).toMatchSnapshot("parameters");
    });

    it("changeSelection() should change the selection", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.changeSelection(changedSelectionWithIrrelevantElements);

        expect(attributeFilterHandler.getWorkingSelection()).toEqual(changedSelectionWithIrrelevantElements);
    });

    it("revertSelection() should trigger onSelectionChanged() callback", async () => {
        const onSelectionChanged = vi.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.changeSelection(changedSelection);

        attributeFilterHandler.onSelectionChanged(onSelectionChanged);
        attributeFilterHandler.revertSelection();

        expect(onSelectionChanged).toHaveBeenCalledTimes(1);
        expect(onSelectionChanged.mock.calls[0]).toMatchSnapshot("parameters");
    });

    it("revertSelection() should revert the selection", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        const selectionBeforeTheChange = attributeFilterHandler.getCommittedSelection();

        attributeFilterHandler.changeSelection(changedSelection);
        attributeFilterHandler.revertSelection();

        expect(attributeFilterHandler.getWorkingSelection()).toEqual(selectionBeforeTheChange);
    });

    it("invertSelection() should trigger onSelectionChanged() callback", async () => {
        const onSelectionChanged = vi.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onSelectionChanged(onSelectionChanged);
        attributeFilterHandler.invertSelection();

        expect(onSelectionChanged).toHaveBeenCalledTimes(1);
        expect(onSelectionChanged.mock.calls[0]).toMatchSnapshot("parameters");
    });

    it("invertSelection() should invert the selection", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        const selectionBeforeTheChange = attributeFilterHandler.getWorkingSelection();

        attributeFilterHandler.invertSelection();

        expect(attributeFilterHandler.getWorkingSelection().isInverted).toBe(
            !selectionBeforeTheChange.isInverted,
        );
    });

    it("commitSelection() should trigger onSelectionCommitted() callback", async () => {
        const onSelectionCommitted = vi.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onSelectionCommitted(onSelectionCommitted);
        attributeFilterHandler.changeSelection(changedSelection);
        attributeFilterHandler.commitSelection();

        expect(onSelectionCommitted).toHaveBeenCalledTimes(1);
        expect(onSelectionCommitted.mock.calls[0]).toMatchSnapshot("parameters");
    });

    it("commitSelection() should commit the selection", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.changeSelection(changedSelection);
        attributeFilterHandler.commitSelection();

        expect(attributeFilterHandler.getCommittedSelection()).toEqual(changedSelection);
    });

    it("clearSelection() should trigger onSelectionChanged() callback", async () => {
        const onSelectionChanged = vi.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onSelectionChanged(onSelectionChanged);
        attributeFilterHandler.clearSelection();

        expect(onSelectionChanged).toHaveBeenCalledTimes(1);
        expect(onSelectionChanged.mock.calls[0]).toMatchSnapshot("parameters");
    });

    it("clearSelection() should clear the selection", async () => {
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.changeSelection(changedSelection);
        attributeFilterHandler.clearSelection();

        expect(attributeFilterHandler.getWorkingSelection().keys).toEqual([]);
    });
});
