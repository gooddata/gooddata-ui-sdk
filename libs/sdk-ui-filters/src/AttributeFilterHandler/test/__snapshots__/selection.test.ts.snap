// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultiSelectAttributeFilterHandler > changeSelection() should trigger onSelectionChanged() callback > parameters 1`] = `
[
  {
    "selection": {
      "irrelevantKeys": [],
      "isInverted": true,
      "keys": [
        "/selectedElement",
      ],
    },
  },
]
`;

exports[`MultiSelectAttributeFilterHandler > clearSelection() should trigger onSelectionChanged() callback > parameters 1`] = `
[
  {
    "selection": {
      "irrelevantKeys": [],
      "isInverted": false,
      "keys": [],
    },
  },
]
`;

exports[`MultiSelectAttributeFilterHandler > commitSelection() should trigger onSelectionCommitted() callback > parameters 1`] = `
[
  {
    "selection": {
      "irrelevantKeys": [],
      "isInverted": true,
      "keys": [
        "/selectedElement",
      ],
    },
  },
]
`;

exports[`MultiSelectAttributeFilterHandler > invertSelection() should trigger onSelectionChanged() callback > parameters 1`] = `
[
  {
    "selection": {
      "irrelevantKeys": [],
      "isInverted": true,
      "keys": [
        "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
        "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
      ],
    },
  },
]
`;

exports[`MultiSelectAttributeFilterHandler > revertSelection() should trigger onSelectionChanged() callback > parameters 1`] = `
[
  {
    "selection": {
      "irrelevantKeys": [],
      "isInverted": false,
      "keys": [
        "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
        "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
      ],
    },
  },
]
`;

exports[`MultiSelectAttributeFilterHandler > should initialize selection for negative attribute filter > committed selection 1`] = `
{
  "irrelevantKeys": [],
  "isInverted": true,
  "keys": [
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
  ],
}
`;

exports[`MultiSelectAttributeFilterHandler > should initialize selection for negative attribute filter > working selection 1`] = `
{
  "irrelevantKeys": [],
  "isInverted": true,
  "keys": [
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
  ],
}
`;

exports[`MultiSelectAttributeFilterHandler > should initialize selection for positive attribute filter > committed selection 1`] = `
{
  "irrelevantKeys": [],
  "isInverted": false,
  "keys": [
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
  ],
}
`;

exports[`MultiSelectAttributeFilterHandler > should initialize selection for positive attribute filter > working selection 1`] = `
{
  "irrelevantKeys": [],
  "isInverted": false,
  "keys": [
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
    "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
  ],
}
`;
